geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/AbstractRedisData.java

```diff
@@ -218,7 +230,14 @@ public void toDelta(DataOutput out) throws IOException {
 
   @Override
   public void fromDelta(DataInput in) throws IOException, InvalidDeltaException {
+    short deltaVersion = DataSerializer.readPrimitiveShort(in);
     DeltaType deltaType = readEnum(DeltaType.class, in);
+
+    if (deltaType.isVersioned() && deltaVersion == getVersion()) {
+      return;
+    }
+    setVersion(deltaVersion);
```

Comment:
DeltaDUnitTest and PartitionedRegionStatsUpdateTest tests were failing because of `setVersion(deltaVersion)` because ValueAppend increments the version. So every other delta was skipped because after `valueAppend` the version was one more than expected. Commenting out the `setVersion` made the DunitTests pass not sure if there are any side effects.

Reply:
Right. So I've pulled the version increment out of that method into the actual `RedisString.append` call. That way the delta application won't cause a problem.